lib/whitespace.coffee

```diff
@@ -31,14 +31,24 @@ class Whitespace
         if atom.config.get('whitespace.ensureSingleTrailingNewline', scope: scopeDescriptor)
           @ensureSingleTrailingNewline(editor)
 
+    editorTextInsertedSubscription = editor.onDidInsertText (event) =>
+      editor.transact =>
```

Comment:
There's a tiny bit of overhead involved with opening and closing a transaction (snapshotting cursor and selection positions), even if no changes are performed. Could you refactor this such that `transact` is not called unless we make changes?